lib/grammers-client/examples/dialogs.rs

```diff
@@ -58,7 +60,29 @@ async fn async_main() -> Result<()> {
         let phone = prompt("Enter your phone number (international format): ")?;
         let token = client.request_login_code(&phone, api_id, &api_hash).await?;
         let code = prompt("Enter the code you received: ")?;
-        client.sign_in(&token, &code).await?;
+        let signed_in = client.sign_in(&token, &code).await;
+        match signed_in {
+            Ok(SignInResult::PasswordRequired {
+                password_information,
+            }) => {
+                // Note: this `prompt` method will echo the password in the console.
+                //       Real code might want to use a better way to handle this.
+                let hint = password_information.hint.as_ref().unwrap();
+                let prompt_message = format!("Enter the password (hint {}): ", &hint);
+                let mut password = prompt(prompt_message.as_str())?.into_bytes();
+
+                // Remove \n symbol if any
+                if password.len() > 0 && password.last().unwrap() == &10u8 {
+                    password.remove(password.len() - 1);
+                }
```

Comment:
Can this actually happen? Would a simple [`str::trim`](https://doc.rust-lang.org/stable/std/primitive.str.html#method.trim) work here, for the sake of the example? I don't think many users are out there with leading or trailing whitespace in their passwords. I don't even think official applications let you enter a new line in a password. Is the newline something our `prompt` returns?